worldedit-core/src/main/java/com/sk89q/worldedit/regions/Polygonal2DRegion.java

```diff
@@ -207,7 +207,7 @@ public int getArea() {
             j = i;
         }
 
-        return (int) Math.floor(Math.abs(area * 0.5)
+        return (long) Math.floor(Math.abs(area * 0.5)

```

Comment:
This should also be changed to `BigDecimal`, and I think the area computation above also needs to be at least promoted to `long` to ensure we don't lose the edge cases:
```
>>> (3_000_000 * 3_000_000) > 2**32
True
```